### PR TITLE
Added functions to utilize the "Random" PKCS#11 functions

### PR DESCRIPTION
**Pull request by mjb3279**
- Added `generate_random_slice` which will put random data into a provided slice
- Added `generate_random_vec` which will return random data in a `Vec<u8>`
- Added `seed_random` which will seed the RNG on the token

Signed-off-by: Michael J Boquard <michael.j.boquard@gmail.com>

### CODE REVIEW

File: cryptoki/src/functions/random.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+//! Functions used to generate random numbers
+
+use crate::get_pkcs11;
+use crate::types::function::Rv;
+use crate::types::session::Session;
+use crate::Result;
+use std::convert::TryInto;
+
+impl<'a> Session<'a> {
+    /// Generates a random number and sticks it in a slice
```
Comment (ionut-arm) [inline cryptoki/src/functions/random.rs:12]: Could you also mention here that the length of the slice is used to decide how many bytes are returned?

File: cryptoki/src/functions/random.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                random_data.as_ptr() as *mut u8,
+                random_data.len().try_into()?,
+            ))
+            .into_result()?;
+        }
+        Ok(())
+    }
+
+    /// Generates random data and returns it as a Vec<u8>
+    pub fn generate_random_vec(&self, random_len: u32) -> Result<Vec<u8>> {
```
Comment (ionut-arm) [inline cryptoki/src/functions/random.rs:26]: I'm guessing no one would ask for more than 4 billion bytes?

**Review by ionut-arm — APPROVED** [review]
Thanks, looks pretty good, only one small request.

File: cryptoki/src/functions/random.rs (hunk 2 above)
Comment (mjb3279) [reply cryptoki/src/functions/random.rs:26]: I would hope not...?  That value is gated by the length parameters to `C_GenerateRandom`.

File: cryptoki/src/functions/random.rs [hunk 3]
```diff
@@ -10,6 +10,11 @@ use std::convert::TryInto;
 
 impl<'a> Session<'a> {
     /// Generates a random number and sticks it in a slice
+    ///
+    /// # Arguments
+    ///
+    /// * `random_slice` - The slice to stick the random data into.  The length of the slice represents
+    /// the number of bytes to obtain from the RBG
     pub fn generate_random_slice(&self, random_data: &mut [u8]) -> Result<()> {
```
Comment (mjb3279) [inline cryptoki/src/functions/random.rs:18]: Updated comment block for `generate_random_slice`

**Review by ionut-arm — APPROVED** [review]

**Pull request merged**